### PR TITLE
Feat/reports

### PR DESCRIPTION
**Pull request by rianV2**

### CODE REVIEW

File: internal/repository/db/base.go [hunk 1]
```diff
@@ -60,7 +60,12 @@ func (m *base[T]) GetConn(ctx context.Context) *gorm.DB {
 
 func (m *base[T]) BuildFilterSort(ctx context.Context, query *gorm.DB, filterParam abstraction.Filter) {
 	for _, filter := range filterParam.Query {
-		query.Where(filter.Field+" = ?", filter.Value)
+
+		if filter.Operator != "" {
```
Comment (mcholismalik) [inline internal/repository/db/base.go:64]: if operator empty its not handled yet ?

File: internal/repository/db/base.go [hunk 2]
```diff
@@ -60,7 +60,12 @@ func (m *base[T]) GetConn(ctx context.Context) *gorm.DB {
...
+			query.Where(filter.Field+" "+filter.Operator+" ?", filter.Value)
+		}
+		// !TODO: why we dont add a full query instead ? so we able to customize the operator
+		// ex: filter.Query = "order_trxs = ?"
```
Comment (mcholismalik) [inline internal/repository/db/base.go:68]: column can be used not only for where condition, its possibly for order by, group by, in next improvement, wdyt ?

File: internal/usecase/report.go [hunk 3]
```diff
@@ -26,62 +28,67 @@ func NewReport(cfg *config.Configuration, f repository.Factory) Report {
...
-				query.StartDateTime,
-				query.EndDateTime,
-			},
-		}
-	}
-
 	orders, info, err := u.Repo.OrderTrx.Find(ctx, filterParam, search)
 	if err != nil {
-		return nil, pagination, res.ErrorBuilder(res.Constant.Error.InternalServerError, err)
+		return dto.OrderReportResponse{}, pagination, res.ErrorBuilder(res.Constant.Error.InternalServerError, err)
```
Comment (mcholismalik) [inline internal/usecase/report.go:35]: return result

File: internal/usecase/report.go [hunk 4]
```diff
@@ -26,62 +28,67 @@ func NewReport(cfg *config.Configuration, f repository.Factory) Report {
...
-		search = &abstraction.Search{
-			Query: searchQuery,
-			Args: []interface{}{
-				query.StartDateTime,
-				query.EndDateTime,
-			},
-		}
+
+	var (
+		orders []entity.OrderViewProduct //!TODO, why this was define as entity.OrderViewProduct not model.OrderViewProduct ?
```
Comment (mcholismalik) [inline internal/usecase/report.go:63]: cz model contains 3 type (entity, dto, abstraction), so model in this boilerplate is general

File: internal/usecase/report.go [hunk 5]
```diff
@@ -26,62 +28,67 @@ func NewReport(cfg *config.Configuration, f repository.Factory) Report {
...
+		info   *abstraction.PaginationInfo
+		count  int64
+	)
+	switch query.Group {
+	case constant.GROUP_BY_VARIANT:
+		logrus.Info("FindGroupByVariant")
+		orders, count, info, err = u.Repo.OrderTrx.FindGroupByVariant(ctx, filterParam, search)
+		break
```
Comment (mcholismalik) [inline internal/usecase/report.go:71]: no need break here

File: internal/usecase/report.go [hunk 6]
```diff
@@ -26,62 +28,67 @@ func NewReport(cfg *config.Configuration, f repository.Factory) Report {
...
+	case constant.GROUP_BY_PACKET:
+		logrus.Info("FindGroupByPacket")
+		orders, count, info, err = u.Repo.OrderTrx.FindGroupByPacket(ctx, filterParam, search)
+		break
```
Comment (mcholismalik) [inline internal/usecase/report.go:75]: no need break here

File: internal/repository/db/base.go (hunk 1 above)
Comment (rianV2) [reply internal/repository/db/base.go:64]: We already have default value for operator [https://github.com/nakoding-community/stoqu-be/blob/feat/reports/internal/model/abstraction/filter.go#L75](here). Assume operator will not empty. Wdyt?

Comment (mcholismalik) [reply internal/repository/db/base.go:64]: I see, seems gud

**Review by mcholismalik — APPROVED** [review]
lgtm

**Pull request merged**